src/bdecode.cpp

```diff
@@ -193,6 +193,13 @@ namespace libtorrent
 		}
 	}
 
+	// diff between current and next item offset
+	inline int tokens_offset_diff(bdecode_token const* t)
```

Comment:
first: there's no need to use pointers here (unless you tell me that the optimizer somehow can optimize a pointer but not a reference). A pointer, semantically, mean that something can be null. a reference is the same thing, but without pointer arithmetic and without being null. It seems like a clear win here. ``t`` cannot be null.

However, you do need the pointer arithmetic here. I suggested you try: ``(&t)[1].offset`` instead of ``(t + 1)->offset``. Did you try that, and did you get a different result? I would be very surprised if it generates different code.

I can't say I'm keen on having a function like this at all, but here are some comments about it:

1. the name does not really describe what it does (at least not as far as I can tell)
2. It should not be ``inline`` (it won't appear in any other translation units)
3. It should have internal linkage (i.e. be in the anonymous namespace)